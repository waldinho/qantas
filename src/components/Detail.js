import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`

`;

const Detail = ({
    airportName,
}) => {
    return(
        <Wrapper key={airportName}>
            <div key={airportName}>
                <p aria-label={airportName}>{airportName} details</p>
            </div>
        </Wrapper>  
    );
}

export default Detail;