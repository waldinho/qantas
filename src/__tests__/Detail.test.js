import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { shallow } from 'enzyme';
import Detail from '../components/Detail';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <Router>
            <Detail />
        </Router>, div);
    ReactDOM.unmountComponentAtNode(div);
})

describe('<Detail /> with no props', () => {
    const container = shallow(
        <Router>
            <Detail />
        </Router>
    );
    it('should match the snapshot', () => {
        expect(container.html()).toMatchSnapshot();
    }),
    it('able to find an html element', () => {
        expect(container.find('h2').length).toBe(0);
        expect(container.find('p').length).toBe(0);
    })
})