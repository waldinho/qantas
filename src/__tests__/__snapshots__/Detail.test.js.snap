// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Detail /> with no props should match the snapshot 1`] = `"<div class=\\"sc-bwzfXH kIFqpD\\"><div class=\\"main\\"><div><h2> - </h2><p>, , </p></div><p aria-label=\\"View Details\\"><a class=\\"link\\" href=\\"/\\">&gt;Back</a></p></div><div>Loading...</div></div>"`;
